contrib/kind-dual-stack-conversion.sh

```diff
@@ -0,0 +1,289 @@
+      -e "s#^  net_cidr.*#&\,${SECONDARY_CLUSTER_SUBNET}#" \
+      -e "s#^  svc_cidr:.*#&\,${SECONDARY_SERVICE_SUBNET}#" \
+  )
+  echo "Patched OVN config:"
+  echo "${FIXED_OVNCONFIG}"
+  printf '%s' "${FIXED_OVNCONFIG}" | kubectl apply -f -
+  # restart ovnkube-master
+  # FIXME: kubectl rollout restart deployment leaves the old pod hanging 
+  # as workaround we delete the master directly
+  kubectl -n ovn-kubernetes delete pod -l name=ovnkube-master
+  # restart ovnkube-node
+  kubectl -n ovn-kubernetes rollout restart daemonset ovnkube-node
+  kubectl -n ovn-kubernetes rollout status daemonset ovnkube-node
+  echo "Updated CNI"
+}
+
+convert_k8s_control_plane(){
+  # Kubeadm installs apiserver and controller-manager as static pods
+  # one the configuration has changed kubelet restart them
+  API_CONFIG_FILE="/etc/kubernetes/manifests/kube-apiserver.yaml"
+  CM_CONFIG_FILE="/etc/kubernetes/manifests/kube-controller-manager.yaml"
+
+  # update all the control plane nodes
+  for n in $CONTROL_PLANE_NODES; do
+    echo "Converting control-plane on node $n"
+    # kube-apiserver backup config file
+    docker exec $n cp ${API_CONFIG_FILE} /kind/${API_CONFIG_FILE}.bak
+    # append second service cidr --service-cluster-ip-range=<IPv4 CIDR>,<IPv6 CIDR>
+    docker exec $n sed -i -e "s#.*service-cluster-ip-range.*#&\,${SECONDARY_SERVICE_SUBNET}#" ${API_CONFIG_FILE}
+    # OVN has already the SCTP feature enable, we just need to append the new one
+    # NOTE: if there was not feature-gate enabled we must use    
+    # docker exec $n sed -i '/service-cluster-ip-range/i\    - --feature-gates=IPv6DualStack=true' ${API_CONFIG_FILE}
+    docker exec $n sed -i -e "s#.*feature-gates.*#&\,IPv6DualStack=true#" ${API_CONFIG_FILE}
+    # kube-controller-manager /etc/kubernetes/manifests/kube-controller-manager.yaml
+    docker exec $n cp ${CM_CONFIG_FILE} /kind/${CM_CONFIG_FILE}.bak
+    # append second service cidr --service-cluster-ip-range=<IPv4 CIDR>,<IPv6 CIDR>
+    docker exec $n sed -i -e "s#.*service-cluster-ip-range.*#&\,${SECONDARY_SERVICE_SUBNET}#" ${CM_CONFIG_FILE}
+    # append second cluster cidr --cluster-cidr=<IPv4 CIDR>,<IPv6 CIDR>
+    docker exec $n sed -i -e "s#.*cluster-cidr.*#&\,${SECONDARY_CLUSTER_SUBNET}#" ${CM_CONFIG_FILE}
+    # OVN has already the SCTP feature enable, we just need to append the new one
+    # NOTE: if there was not feature-gate enabled we must use    
+    # docker exec $n sed -i '/service-cluster-ip-range/i\    - --feature-gates=IPv6DualStack=true' ${CM_CONFIG_FILE}
+    docker exec $n sed -i -e "s#.*feature-gates.*#&\,IPv6DualStack=true#" ${CM_CONFIG_FILE}
+    echo "Finished converting control plane on node $n"
+  done
+
+}
+
+usage()
+{
+    echo "usage: kind_dual_conversion.sh [-n|--name <cluster_name>] [-ss secondary_service_subnet] [-sc secondary_cluster_subnet]"
+    echo "Convert a single stack cluster in a dual stack cluster"
+}
+
+parse_args()
+{
+    while [ "$1" != "" ]; do
```

Comment:
to avoid unbound variable here, you can move the set -eux to after parse_args like we do in kind.sh: https://github.com/ovn-org/ovn-kubernetes/blob/master/contrib/kind.sh#L263

Reply:
could also add pipefail:
set -euxo pipefail

Reply:
done